duckduckgo/chrome-zeroclickinfo
# Port base framework over from SERP, build with grunt+babel+browserify

https://app.asana.com/0/312783424879085/319083062283455

For posterity, I also did a short write-up about what's included here:
https://dub.duckduckgo.com/gist/lauren/f75de306be9fe8f607da4a35706a8cca

**Reviewer:**
@russellholt 



## Description:
This is the port of the SERP framework over. Lots of big changes. Wondering what the best venue is for details on this. Highlights:

- We are now using Grunt to build ES6 (aka "next" version of Javascript) files that have the extension *.es6.js, transform them back into ES5 (old, browser-friendly javascript), and then using Browserify to analyze and bundle them into public/js/base.js and public/js/trackers.js, the javascript entry point for trackers.html. For an example of how the SERP framework is supposed to work, check out /js/pages/trackers.js and follow the code.

- Instead of Handlebars, the rendering engine is now upgraded to using an npm package called `bel` so that we can just use template literals, which give us all the power of using vanilla javascript INSIDE of template files. Woo!

## Steps to test this PR:
1. npm install
2. npm run dev (will build the *.es6.js files in addition to previous build tasks from before)
3. navigate to trackers.html, see sub-view "Whitelist" render on page. click on a list item and you'll see the Whitelist view's click handler method console.log a message.

## Automated tests:

###### Reviewer Checklist:


###### PR Author Checklist:
